osquery/remote/http_client.cpp

```diff
@@ -240,6 +240,18 @@ void Client::encryptConnection() {
   }
 
   ssl_sock_ = std::make_shared<ssl_stream>(sock_, ctx);
+
+  if (client_options_.always_verify_peer_) {
+    if (!client_options_.sni_hostname_) {
+      throw std::logic_error(
+          "The SNI hostname is missing from the Client options");
```

Comment:
Why the capital `Client`?

Reply:
It is referring to the Client class, this ideally never happens unless the code forgot to set the proper values. Should I make it lowercase?